src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/SwitchEditorCard.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { injectIntl, FormattedMessage, intlShape } from '@edx/frontend-platform/i18n';
+import { Card } from '@openedx/paragon';
+import PropTypes from 'prop-types';
+import messages from '../messages';
+import { selectors, thunkActions } from '../../../../../../data/redux';
+import BaseModal from '../../../../../../sharedComponents/BaseModal';
+import Button from '../../../../../../sharedComponents/Button';
+import { handleConfirmEditorSwitch } from '../hooks';
+import { ProblemTypeKeys } from '../../../../../../data/constants/problem';
+
+const SwitchEditorCard = ({
+  intl,
+  editorType,
+  problemType,
+  switchEditor,
+  isMarkdownEditorEnabled,
+}) => {
+  const [isConfirmOpen, setConfirmOpen] = React.useState(false);
+
+  if (isMarkdownEditorEnabled || problemType === ProblemTypeKeys.ADVANCED) { return null; }
+
+  return (
+    <Card className="border border-light-700 shadow-none">
+      <BaseModal
+        isOpen={isConfirmOpen}
+        close={() => { setConfirmOpen(false); }}
+        title={intl.formatMessage(messages.ConfirmSwitchMessageTitle, { convertType: editorType === 'advanced' ? 'OLX' : 'Markdown' })}
+        confirmAction={(
+          <Button
+            onClick={() => handleConfirmEditorSwitch({ switchEditor: () => switchEditor(editorType), setConfirmOpen })}
+            variant="primary"
+          >
+            {intl.formatMessage(messages.ConfirmSwitchButtonLabel, { editorType })}
+          </Button>
+        )}
+        size="md"
+      >
+        {editorType === 'advanced' ? <FormattedMessage {...messages.ConfirmSwitcToAdvancedMessage} /> : <FormattedMessage {...messages.ConfirmSwitchToMardownMessage} />}
+      </BaseModal>
+      <Button
+        className="my-3 ml-2 py-0"
+        variant="link"
+        size="sm"
+        onClick={() => { setConfirmOpen(true); }}
+      >
+        {intl.formatMessage(messages.SwitchButtonLabel, { editorType }) }
+      </Button>
+    </Card>
+  );
+};
+
+SwitchEditorCard.propTypes = {
+  intl: intlShape.isRequired,
+  switchEditor: PropTypes.func.isRequired,
+  isMarkdownEditorEnabled: PropTypes.bool.isRequired,
+  problemType: PropTypes.string.isRequired,
+  editorType: PropTypes.string.isRequired,
+};
+
+export const mapStateToProps = (state) => ({
+  isMarkdownEditorEnabled: selectors.problem.isMarkdownEditorEnabled(state)
+     && selectors.app.isMarkdownEditorEnabledForCourse(state),
+});
+export const mapDispatchToProps = {
+  switchEditor: thunkActions.problem.switchEditor,
+};
+
+export const SwitchEditorCardInternal = SwitchEditorCard; // For testing only
+export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(SwitchEditorCard));
```

Comment:
I realize this file is just renamed, and not a new file, so consider this optional / non-blocking: 

`injectIntl` is deprecated. Please use `useIntl` instead. https://github.com/openedx/frontend-app-learning/pull/1634

Also, please don't use `connect` or really any other higher order components like this - instead of `mapStateToProps` and `mapDispatchToProps`, use the `useSelector` and `useDispatch` hooks respectively.

Once you do this you can likely get rid of `SwitchEditorCardInternal` as it won't be needed.

(Fine to leave in existing code, but new code should follow the new practices)